File: ruby/hamming/hamming.rb

module Hamming
  def Hamming.compute(dna1, dna2)
    if dna1.size != dna2.size
      raise ArgumentError, "DNA strands size must be equal"
    end
    index = 0
    diff_nucleotides = 0
    dna1.each_char do |c|
      if c != dna2[index]
        diff_nucleotides += 1
      end
      index += 1
    end
    diff_nucleotides
  end
end

File: lua/hamming/hamming.lua

local m = {}

function m.compute(dna1, dna2)
  if string.len(dna1) ~= string.len(dna2) then
    return -1
  end
  local diff = 0
  local i = 1
  for c in dna1:gmatch(".") do
    if c ~= dna2:sub(i, i) then
      diff = diff + 1
    end
    i = i + 1
  end
  return diff
end

return m
